DEVELOPMENT.md

```diff
@@ -98,6 +98,69 @@ make lint
 > [!NOTE]
 > Besides the linters mentioned above, `golangci-lint` is also executed for [pull requests](.github/workflows/ci-pre-merge.yml).
 
+## Protocol Buffers
+
+When you need to make changes to the protocol buffer messages and/or services, change the `.proto` files, and then run `make generate` to update the files generated from the `.proto` files.
+
+To run `make generate`, install [Buf](https://buf.build/) and plugins specified in [the CLI configuration files](https://buf.build/docs/configuration/v1/buf-yaml/) as follows.
+
+When contributing, please use the same version of tools/plugins as the ones specified in the example commands below.
+Otherwise, there will be unnecessary diffs in the generated files because of version differences.
+
+### Buf
+
+We're using Buf instead of [`protoc`](https://github.com/protocolbuffers/protobuf?tab=readme-ov-file#protobuf-compiler-installation).
+
+Install it by:
+
+```console
+go install github.com/bufbuild/buf/cmd/buf@v1.5.0
```

Comment:
Buf `v1.32.0` and later require `buf.lock` to have the digest field. When bumping to newer versions, get the field generated by some older version.